src/sqlfluff/dialects/dialect_athena.py

```diff
@@ -643,3 +646,62 @@ class IntervalExpressionSegment(BaseSegment):
             ),
         ),
     )
+
+
+class GroupByClauseSegment(ansi.GroupByClauseSegment):
+    """A `GROUP BY` clause like in `SELECT`.
+
+    https://docs.aws.amazon.com/athena/latest/ug/select.html#:~:text=%5B-,GROUP,-BY%20%5B%20ALL%20%7C%20DISTINCT%20%5D%20grouping_expressions
+    """
+
+    match_grammar: Matchable = Sequence(
+        "GROUP",
+        "BY",
+        Indent,
+        Delimited(
+            OneOf(
+                Ref("CubeRollupClauseSegment"),
+                Ref("GroupingSetsClauseSegment"),
+                Ref("ColumnReferenceSegment"),
+                Ref("NumericLiteralSegment"),  # Can `GROUP BY 1`
+                Ref("ExpressionSegment"),  # Can `GROUP BY coalesce(col, 1)`
+            ),
+            terminator=OneOf(
+                Sequence("ORDER", "BY"),
+                "LIMIT",
+                "OFFSET",
+                "HAVING",
+                Ref("SetOperatorSegment"),
+            ),
+        ),
+        Dedent,
+    )
+
+
+class CubeRollupClauseSegment(BaseSegment):
+    """`[CUBE | ROLLUP]` clause within the `GROUP BY` clause."""
+
+    type = "cube_rollup_clause"
+
+    match_grammar = Sequence(
+        OneOf("CUBE", "ROLLUP"),
+        Bracketed(Delimited(Ref("ColumnReferenceSegment"))),
+    )
+
+
+class GroupingSetsClauseSegment(BaseSegment):
+    """`GROUPING SETS` clause within the `GROUP BY` clause."""
+
+    type = "grouping_sets_clause"
+
+    match_grammar = Sequence(
+        "GROUPING",
+        "SETS",
+        Bracketed(
+            Delimited(
+                Ref("ColumnReferenceSegment"),
+                Bracketed(Delimited(Ref("ColumnReferenceSegment"))),
```

Comment:
```suggestion
                Bracketed(Delimited(Ref("ColumnReferenceSegment"), optional=True)),
```
And then can you remove the `Bracketed()` below?